### PR TITLE
Add object detection documentation

### PR DESCRIPTION
**Pull request by alonfaraj**
Add object detection documentation for this PR https://github.com/microsoft/AirSim/pull/3472

### CODE REVIEW

File: docs/object_detection.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Object Detection
+
+## About
+This feature let you generate object detection using exiting cameras in AirSim, similar to detection DNN.
```
Comment (rajat2004) [inline docs/object_detection.md:4]: exiting -> existing

File: docs/object_detection.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+## About
+This feature let you generate object detection using exiting cameras in AirSim, similar to detection DNN.   
+Using the API you can control which object to detect by name and radius from camera.   
+One can control these settings for each camera, image type and vehicle combination separately.
+
+## API
+- Set mesh name  to detect in wildcard format   
+```simAddDetectionFilterMeshName(camera_name, image_type, mesh_name, vehicle_name = '')```   
+
+- Clear all mesh names previosly added
```
Comment (rajat2004) [inline docs/object_detection.md:12]: previously

File: docs/object_detection.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    'longitude': 34.864785008379876},
+    'name': 'Cylinder9_2',
+    'relative_pose': <Pose> {   'orientation': <Quaternionr> {   'w_val': 0.9929741621017456,
+    'x_val': 0.0038591264747083187,
+    'y_val': -0.11333247274160385,
+    'z_val': 0.03381215035915375},
+    'position': <Vector3r> {   'x_val': 4.400000095367432,
+    'y_val': 0.29999998211860657,
+    'z_val': 1.0199999809265137}}}
+ ```
```
Comment (rajat2004) [inline docs/object_detection.md:68]: Extra space before backtick causes problems in final doc

File: docs/object_detection.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+Python script [detection.py](https://github.com/microsoft/AirSim/blob/master/PythonClient/detection/detection.py) shows how to set detection parameters and show the result in OpenCV capture.
+
+A miniml example using API with Blocks enviromnet:
+```python
+camera_name = "0"
+image_type = airsim.ImageType.Scene
+
+client.simSetDetectionFilterRadius(camera_name, image_type, 80 * 100) # in [cm]
+client.simAddDetectionFilterMeshName(camera_name, image_type, "Cylinder_*") 
+client.simGetDetections(camera_name, image_type)
```
Comment (rajat2004) [inline docs/object_detection.md:41]: Maybe something like `detections = client.simGetDetections(...)`?

File: docs/object_detection.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    'x_val': 0.0038591264747083187,
+    'y_val': -0.11333247274160385,
+    'z_val': 0.03381215035915375},
+    'position': <Vector3r> {   'x_val': 4.400000095367432,
+    'y_val': 0.29999998211860657,
+    'z_val': 1.0199999809265137}}}
+ ```
+ 
+ ![113001312-5babfb00-9179-11eb-926e-cd6e562c92ac](https://user-images.githubusercontent.com/15960497/121812071-335b7500-cc6f-11eb-8914-c5288e1cbb47.png)
+![113001322-5d75be80-9179-11eb-8d3f-443b68f7cba7](https://user-images.githubusercontent.com/15960497/121812072-348ca200-cc6f-11eb-9032-b79cbb672cf0.png)
```
Comment (rajat2004) [inline docs/object_detection.md:71]: Better to put images in the `images/` folder itself rather than linking to a comment

**Review by rajat2004 — COMMENTED** [review]
Also need to add the page to `mkdocs.yml` present in the root folder so that the page appears in the side navbar. You can use `mkdocs serve` to see the final page

**Comment by alonfaraj** [comment]
@rajat2004 Thank you very much for the review and guidance!

File: docs/object_detection.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    name = ''
+    geo_point = GeoPoint()
+    box2D = Box2D()
+    box3D = Box3D()
+    relative_pose = Pose()
+```
+## Usage example
+Python script [detection.py](https://github.com/microsoft/AirSim/blob/master/PythonClient/detection/detection.py) shows how to set detection parameters and show the result in OpenCV capture.
+
+A minimal example using API with Blocks enviromnet to detect Cylinder objects:
```
Comment (zimmy87) [inline docs/object_detection.md:34]: `enviromnet` should be `environment`

File: docs/object_detection.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
+# Object Detection
+
+## About
+This feature let you generate object detection using existing cameras in AirSim, similar to detection DNN.
```
Comment (zimmy87) [inline docs/object_detection.md:4]: `This feature let you` should be `This feature lets you`

File: docs/object_detection.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
+# Object Detection
+
+## About
+This feature let you generate object detection using existing cameras in AirSim, similar to detection DNN.   
+Using the API you can control which object to detect by name and radius from camera.   
+One can control these settings for each camera, image type and vehicle combination separately.
+
+## API
+- Set mesh name  to detect in wildcard format
```
Comment (zimmy87) [inline docs/object_detection.md:9]: there's an extra space between `mesh name` and `to`

File: docs/object_detection.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+```python
+DetectionInfo
+    name = ''
+    geo_point = GeoPoint()
+    box2D = Box2D()
+    box3D = Box3D()
+    relative_pose = Pose()
+```
+## Usage example
+Python script [detection.py](https://github.com/microsoft/AirSim/blob/master/PythonClient/detection/detection.py) shows how to set detection parameters and show the result in OpenCV capture.
```
Comment (zimmy87) [inline docs/object_detection.md:32]: `and show` should be `and shows`

**Review by zimmy87 — CHANGES_REQUESTED** [review]
Just a couple typos

**Review by zimmy87 — APPROVED** [review]

**Comment by zimmy87** [comment]
The MacOS build check failure is a known issue already fixed in master, so it should disappear after merging. All CR feedback has been addressed so I am moving ahead with merge.

**Pull request merged**